docs/common/ethereum-definitions.md

```diff
@@ -0,0 +1,165 @@
+# Ethereum definitions
+
+For support of the huge number of EVM chains (networks) and ERC-20 tokens, Trezor needs
+to know parameters of those networks and tokens, namely:
+
+* currency symbol and number of decimal places, to correctly display amounts,
+* SLIP44 identifier to unlock the appropriate BIP-32 subtrees.
+
+A subset of Ethereum definitions is built into the firmware image. The rest is generated
+externally and must be sent to Trezor as a signed blob.
+
+## Built-in definitions
+
+The set of built-in definitions is declared in the following files:
+* networks - [`networks.json`](https://github.com/trezor/trezor-firmware/blob/master/common/defs/ethereum/networks.json)
+* tokens - [`tokens.json`](https://github.com/trezor/trezor-firmware/blob/master/common/defs/ethereum/tokens.json)
+
+These definitions need to be modified manually.
+
+## External definitions
+
+A full list of Ethereum definitions is compiled from multiple sources and is available
+[in a separate repository](https://github.com/trezor/definitions).
+
+From this list, a collection of binary blobs is generated, signed, and made available
+online.
+
+A given Trezor firmware will only accept signed definitions newer than a certain date,
+typically one month before firmware release. This means that a client application should
+either always fetch fresh definitions from the official URLs, or refresh its local copy
+frequently.
+
+### Retrieving the definitions
+
+The base URL for the definitions is `https://data.trezor.io/firmware/eth-definitions/`.
+
+#### Known chain ID
+
+To look up a network definition by its chain ID, use the following URL:
+
+`https://data.trezor.io/firmware/eth-definitions/chain-id/<CHAIN_ID>/network.dat`
+
+`<CHAIN_ID>` is a decimal number, e.g., `1` for Ethereum mainnet.
+
+To look up a token definition for a given chain ID and token address, use the following URL:
+
+`https://data.trezor.io/firmware/eth-definitions/chain-id/<CHAIN_ID>/token-<TOKEN_ADDRESS>.dat`
+
+`<CHAIN_ID>` is again a decimal number.<br>
+`<TOKEN_ADDRESS>` is all lowercase token address hex without the `0x` prefix.
+
+E.g., this is the URL for Görli TST token:
+[https://data.trezor.io/firmware/eth-definitions/chain-id/5/token-7af963cf6d228e564e2a0aa0ddbf06210b38615d.dat]
```

Comment:
this does not work 
https://data.trezor.io/firmware/eth-definitions/chain-id/5/token-7af963cf6d228e564e2a0aa0ddbf06210b38615d.dat

https://data.trezor.io/firmware/eth-definitions/by_chain_id/5/token_7af963cf6d228e564e2a0aa0ddbf06210b38615d.dat

Reply:
expected behavior. also the version on your old-style URL will be rejected by current code in this branch.
i'll try to get the new links up ASAP